src/bosh_azure_cpi/spec/unit/cloud_spec.rb

```diff
@@ -189,8 +194,28 @@
       end
     end
 
+    context 'when instance_type is not provided in resource_pool' do
+      it 'should raise an error' do
+        expect {
+          cloud.create_vm(
+            agent_id,
+            stemcell_id,
+            {},
+            networks_spec,
+            disk_locality,
+            environment
+          )
+        }.to raise_error("missing required cloud property `instance_type` in resource_pool.")
```

Comment:
I would not mention "resource pool". That's a concept in the Director (not the CPI) and we are already moving away from it since we now have VM types.

Also use ' instead of `.